src/main/kotlin/com/team4099/robot2023/subsystems/drivetrain/drive/Drivetrain.kt

```diff
@@ -272,35 +292,31 @@ class Drivetrain(val gyroIO: GyroIO, swerveModuleIOs: DrivetrainIO) : SubsystemB
   }
 
   private fun updateOdometry() {
-    val wheelDeltas =
-      mutableListOf(
-        SwerveModulePosition(),
-        SwerveModulePosition(),
-        SwerveModulePosition(),
-        SwerveModulePosition()
-      )
-    for (i in 0 until 4) {
-      wheelDeltas[i] =
-        SwerveModulePosition(
-          swerveModules[i].inputs.drivePosition.inMeters -
-            lastModulePositions[i].distanceMeters,
-          swerveModules[i].inputs.steeringPosition.inRotation2ds
-        )
-      lastModulePositions[i] =
-        SwerveModulePosition(
-          swerveModules[i].inputs.drivePosition.inMeters,
-          swerveModules[i].inputs.steeringPosition.inRotation2ds
-        )
+
+    var deltaCount =
+      if (gyroInputs.gyroConnected) gyroInputs.odometryYawPositions.size else Integer.MAX_VALUE
+    for (i in 0..4) {
+      deltaCount = Math.min(deltaCount, swerveModules[i].positionDeltas.size)
     }
+    for (deltaIndex in 0..deltaCount) {
```

Comment:
please to avoid making changes that are not part of this pr. make a new pr if needed, doing stuff like this makes it hard to merge in the end